beeline.go

```diff
@@ -162,6 +198,11 @@ func Init(config Config) {
 	// add a bunch of fields
 	if config.ServiceName != "" {
 		client.AddField("service_name", config.ServiceName)
+		client.AddField("service.name", config.ServiceName)
```

Comment:
Hm, I don't think it's being added twice as it's `service_name` vs `service.name`, but I updated for clarity, including some code comments and a change to this block.
`service_name` is added for both classic keys (if non-null) and new keys. `service.name` is added for new keys to more closely mirror OTel.

Reply:
After discussing with the team offline, decided we're going to add fields `service_name` and `service.name` for both classic keys and new keys. This will help in mixed environments, migrations, etc. We may deprecate `service_name` at some point in the future, but in the meantime we will send both.